### PR TITLE
UI :- Added permission roles on ingestion actions

### LINKED ISSUE #8936: UI: Implement roles and policy for ingestion actions
image

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes :

- Issue : Closes #8936
-  Added permission roles on ingestion actions

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx [hunk 1]
```diff
@@ -95,6 +102,24 @@ const Ingestion: React.FC<IngestionProps> = ({
     setSearchText(searchValue);
   };
 
+  const [list, setList] = useState<Array<IngestionPipeline>>([]);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:105]: @Ashish8689 , Are we not getting ingestionList in the props?

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx [hunk 2]
```diff
@@ -95,6 +102,24 @@ const Ingestion: React.FC<IngestionProps> = ({
...
+        const response = await getEntityPermissionByFqn(
+          ResourceEntity.INGESTION_PIPELINE,
+          item.name
+        );
+        setServicePermission((prev) => ({ ...prev, [item.name]: response }));
+      } catch (error) {
+        showErrorToast(error as AxiosError);
+      }
+    });
+  };
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:121]: @Ashish8689 , I would recommend using the `Promise.allSettled` here.

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx [hunk 3]
```diff
@@ -302,18 +328,28 @@ const Ingestion: React.FC<IngestionProps> = ({
...
+
+  useEffect(() => {
+    getSearchedIngestions();
   }, [searchText, ingestionList]);
 
+  useEffect(() => {
+    fetchServicePermission();
+  }, [list]);
+
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:352]: Why are we chnaging the `getSearchedIngestions` function and storing the value in setList?
Every time on search if the list value changes then again we are fetching the permission why?

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx [hunk 4]
```diff
@@ -345,6 +384,10 @@ const Ingestion: React.FC<IngestionProps> = ({
       return (
         <Button
           data-testid="deploy"
+          disabled={
+            !isRequiredDetailsAvailable ||
+            !servicePermission?.[ingestion.name]?.EditAll
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:389]: We are using this same thing in multiple places, can we optimise it?
```
!servicePermission?.[ingestion.name]?.EditAll
```

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:389]: @Sachin-chaurasiya  updating it!

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx (hunk 2 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:121]: @Sachin-chaurasiya  ok, will update this.

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx (hunk 3 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx:352]: @Sachin-chaurasiya  We are invoking the `getSearchedIngestions` function multiple times in the code, which is not required, as we set the value in state and use it wherever we want.
For the `Permission` one updating the code.

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx]: Okay, make sure we are testing the search function properly.

File: openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx (hunk 1 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/Ingestion/Ingestion.component.tsx]: @Sachin-chaurasiya we, are updating it.

**Review by chirag-madlani — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**